Modules/System Tests/Azure Blob Services API/src/ABSBlobClientTest.Codeunit.al

```diff
@@ -611,6 +611,68 @@ codeunit 132920 "ABS Blob Client Test"
+        NodeList: XmlNodeList;
+        NextMarker: Text;
+    begin
+        // [SCENARIO] Parse the BLOB storage API response listing BLOBs in a container without the hierarchical namespace
+
+        // [GIVEN] Prepare an XML response from the BLOB Storage API, listing all BLOBs in a container.
+        // [GIVEN] Container does not have hierarchical namespace and contains a single BLOB with the name like "folder/blob"
+        NodeList := ABSHelperLibrary.CreateBlobNodeListFromResponse(ABSTestLibrary.GetServiceResponseBlobWithHierarchicalName(), NextMarker);
+
+        // [WHEN] Invoke BlobNodeListToTempRecord
+        ABSHelperLibrary.BlobNodeListToTempRecord(NodeList, ABSContainerContent);
+
+        // [THEN] "ABS Container Content" contains one record "folder" and one record "blob"
+        Assert.RecordCount(ABSContainerContent, 2);
+
+        ABSContainerContent.SetRange(Name, ABSTestLibrary.GetSampleResponseRootDirName());
+        Assert.RecordCount(ABSContainerContent, 1);
+
+        ABSContainerContent.SetRange(Name, ABSTestLibrary.GetSampleResponseFileName());
+        Assert.RecordCount(ABSContainerContent, 1);
+    end;
+
+    [Test]
+    procedure ParseResponseFromStorageHierarachicalNamespace()
+    var
+        ABSContainerContent: Record "ABS Container Content";
+        ABSHelperLibrary: Codeunit "ABS Helper Library";
+        NodeList: XmlNodeList;
+        NextMarker: Text;
+    begin
+        // [SCENARIO] Parse the BLOB storage API response listing BLOBs in a container with the hierarchical namespace enabled
+
+        // [GIVEN] Prepare an XML response from the BLOB Storage API, listing all BLOBs in a container.
+        // [GIVEN] Container has the hierarchical namespace enabled, and contains a root folder named "rootdir".
+        // [GIVEN] There is a subdirectory "subdir" in the root, and one blob named "blob" in the subdirectory.
+        NodeList := ABSHelperLibrary.CreateBlobNodeListFromResponse(ABSTestLibrary.GetServiceResponseHierarchicalNamespace(), NextMarker);
+
+        // [WHEN] Invoke BlobNodeListToTempRecord
+        ABSHelperLibrary.BlobNodeListToTempRecord(NodeList, ABSContainerContent);
+
+        // [THEN] "ABS Container Content" contains two records for folders and one record "blob"
+        Assert.RecordCount(ABSContainerContent, 3);
+
+        VerifyContinerContentType(ABSContainerContent, ABSTestLibrary.GetSampleResponseRootDirName(), Enum::"ABS Blob Resource Type"::Directory);
+        VerifyContinerContentType(ABSContainerContent, ABSTestLibrary.GetSampleResponseSubdirName(), Enum::"ABS Blob Resource Type"::Directory);
+        VerifyContinerContentType(ABSContainerContent, ABSTestLibrary.GetSampleResponseFileName(), Enum::"ABS Blob Resource Type"::File);
+    end;
+
+    local procedure VerifyContinerContentType(var ABSContainerContent: Record "ABS Container Content"; BlobName: Text[2048]; ExpectedResourceType: Enum "ABS Blob Resource Type")
```

Comment:
There's one small typo. Should be `VerifyContainerContentType`

Reply:
Thanks for catching this! Updated.